client/scraper/general_metadata.py

```diff
@@ -1,21 +1,59 @@
-from HTMLParser import HTMLParser
-import json
-
 from metadata import Metadata
+from client.constants import JsonFields, MetadataFields
+from bs4 import BeautifulSoup
+
 
+class GeneralMetadata(Metadata):
+    prop_map = Metadata.prop_map
 
-class GeneralMetadata(Metadata, HTMLParser):
-    json_return = ""
-    prop_map = {}
+    def __init__(self):
+        Metadata.__init__(self)
 
     def parse_content(self, content):
-        self.feed(content.content)
+        soup = BeautifulSoup(content.content)
+        title = self.fetch_title(soup)
+        desc = self.fetch_desc(soup)
+        image_url = self.fetch_image_url(soup)
+
+        if title:
+            self.prop_map[JsonFields.TITLE] = title
+
+        if desc:
+            self.prop_map[JsonFields.DESC] = desc
+
+        if image_url:
+            media_list = {}
+            media_list[JsonFields.COUNT] = 1
+            media_list[JsonFields.DATA] = []
+            data = {
+                JsonFields.TYPE: JsonFields.IMAGE,
+                JsonFields.SRC: image_url
+            }
+            media_list[JsonFields.DATA].append(data)
+            self.prop_map[JsonFields.MEDIA] = media_list
         return self.to_json(self.prop_map)
```

Comment:
I think we should format this to something like:
<code>
def __init__(self, content):
   Metadata.__init__(self);
   parse_content(content);

def get_json(self):
    return json.dumps(prop_map)
</code>
calling the code would look something like:
<code>
scraper = create_scraper(response)
scraper.get_json()
</code>

basically we want to optimize for extensibility. So make it so that parse_content only creates a metadata object.
Then we dump it as a json when we want the json. This way we can extend into using different data types as return in the future. Also we can send this object to be created in the database in the future.